### PR TITLE
[Remove] Types from DocWrite Request and Response

### PR DESCRIPTION
**Pull request by nknize**
Removes type support from DocWrite Request and Response, all derived classes,
and all places used.

relates #1940

### CODE REVIEW

File: server/src/main/java/org/opensearch/action/delete/DeleteRequestBuilder.java [hunk 1]
```diff
@@ -55,9 +55,10 @@ public DeleteRequestBuilder(OpenSearchClient client, DeleteAction action, @Nulla
 
     /**
      * Sets the type of the document to delete.
+     * @deprecated types will be removed
      */
+    @Deprecated
     public DeleteRequestBuilder setType(String type) {
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/delete/DeleteRequestBuilder.java:61]: This is kept to next pull request?

Comment (nknize) [reply server/src/main/java/org/opensearch/action/delete/DeleteRequestBuilder.java:61]: Yes

File: server/src/main/java/org/opensearch/client/Requests.java [hunk 2]
```diff
@@ -97,8 +97,7 @@ public static IndexRequest indexRequest() {
     }
 
     /**
-     * Create an index request against a specific index. Note the {@link IndexRequest#type(String)} must be
-     * set as well and optionally the {@link IndexRequest#id(String)}.
+     * Create an index request against a specific index.
```
Comment (reta) [inline server/src/main/java/org/opensearch/client/Requests.java:100]:
```suggestion
Note that setting {@link IndexRequest#id(String)} is optional.
```

File: server/src/main/java/org/opensearch/client/Requests.java [hunk 3]
```diff
@@ -109,8 +108,7 @@ public static IndexRequest indexRequest(String index) {
     }
 
     /**
-     * Creates a delete request against a specific index. Note the {@link DeleteRequest#type(String)} and
-     * {@link DeleteRequest#id(String)} must be set.
+     * Creates a delete request against a specific index.
```
Comment (reta) [inline server/src/main/java/org/opensearch/client/Requests.java:111]:
```suggestion
Note that  {@link IndexRequest#id(String)} must be set.
```

File: server/src/test/java/org/opensearch/action/delete/DeleteRequestTests.java [hunk 4]
```diff
@@ -42,23 +42,15 @@
...
         {
-            final DeleteRequest request = new DeleteRequest("index4", "_doc", "0");
+            final DeleteRequest request = new DeleteRequest("index4", "0");
             final ActionRequestValidationException validate = request.validate();
 
             assertThat(validate, nullValue());
         }
 
         {
             // Empty types are accepted but fail validation
```
Comment (reta) [inline server/src/test/java/org/opensearch/action/delete/DeleteRequestTests.java:52]: Probably comment is out of date

**Review by reta — APPROVED** [review]

File: server/src/test/java/org/opensearch/action/delete/DeleteRequestTests.java (hunk 4 above)
Comment (nknize) [reply server/src/test/java/org/opensearch/action/delete/DeleteRequestTests.java]: :+1:

File: server/src/main/java/org/opensearch/client/Requests.java (hunk 3 above)
Comment (nknize) [reply server/src/main/java/org/opensearch/client/Requests.java:112]: Good catch. Done

File: server/src/main/java/org/opensearch/client/Requests.java (hunk 2 above)
Comment (nknize) [reply server/src/main/java/org/opensearch/client/Requests.java:100]: Good catch. Done.

**Review by CEHENKLE — APPROVED** [review]

**Pull request merged**